### ISSUE TITLE
`svgPathLoader` causes webpack to log warnings when no app/svgs directory exists

### ISSUE THREAD

**Issue by benchristel** [pivotal-cf/pivotal-ui#556]
## What version of Pivotal UI are you using?

9.1.2

## What browser and version are you using?

Not relevant, since this is a dev tooling issue.

## What behavior are you seeing?

When I run tests using Karma with the webpack plugin, I see warnings logged about a `require` in Pivotal UI:

```
WARNING in ./~/pivotal-ui/react/iconography/iconography.js
Module not found: Error: Can't resolve '../../../../app/svgs' in '/Users/pivotal/go/src/gpcc/frontend/node_modules/pivotal-ui/react/iconography'
 @ ./~/pivotal-ui/react/iconography/iconography.js 58:17-245
 @ ./~/pivotal-ui/react/iconography/index.js
 @ ./~/pivotal-ui/react/alerts/alerts.js
 @ ./~/pivotal-ui/react/alerts/index.js
 @ ./test/spec/components/admin/myComponent.js

WARNING in ./~/pivotal-ui/react/svg/svg.js
Module not found: Error: Can't resolve '../../../../app/svgs' in '/Users/pivotal/go/src/gpcc/frontend/node_modules/pivotal-ui/react/svg'
 @ ./~/pivotal-ui/react/svg/svg.js 66:13-241
 @ ./~/pivotal-ui/react/svg/index.js
 @ ./~/pivotal-ui/react/iconography/iconography.js
 @ ./~/pivotal-ui/react/iconography/index.js
 @ ./~/pivotal-ui/react/alerts/alerts.js
 @ ./~/pivotal-ui/react/alerts/index.js
 @ ./test/spec/components/admin/myComponent.js
```

## What do you expect to see?

No warnings.

I think this might be happening because webpack is resolving the `require` in `svgPathLoader` at compile time. I've verified that `svgPathLoader` is never actually called; in fact, my tests don't create any Svg elements at all.

For now I am suppressing the warning by creating a dummy `app/svgs/index.js` file.

**Comment by reid47** [comment]
Hi @benchristel, this is the same problem as reported in #549.

It looks like we fixed this for 9.2.1. Please try upgrading to at least that version (and removing the dummy `app/svgs/index.js` file, to ensure it's really working). If you still see these warnings, please let us know.

**Comment by benchristel** [comment]
I've upgraded to 9.2.1. I'm now seeing only one warning instead of two, but it still shows up:

```
WARNING in ./~/pivotal-ui/react/iconography/iconography.js
Module not found: Error: Can't resolve '../../../../app/svgs' in '/Users/pivotal/go/src/gpcc/frontend/node_modules/pivotal-ui/react/iconography'
 @ ./~/pivotal-ui/react/iconography/iconography.js 58:17-245
 @ ./~/pivotal-ui/react/iconography/index.js
 @ ./src/components/admin/permissions.js
 @ ./src/components/admin/index.js
 @ ./src/index.js
```

Which commit was supposed to fix this? I found c1d14aeb, which looks related, but it's also in 9.1.2. There aren't any more commits on that file until 10.0.

**Comment by reid47** [comment]
@benchristel This was the change to `svg.js` that we thought fixed it: https://github.com/pivotal-cf/pivotal-ui/commit/2415cf82198aec57b4434fb64a670d6885c77b8a

I see now that should have made a similar change to `iconography.js` - we only fixed it in `svg.js` because `iconography.js` was rewritten differently for v12 and up, and then we backported only that change.

If you are able to upgrade to v12, you shouldn't see this anymore. Otherwise, we can create a story to backport the fix to v9 and do a patch release.

Thanks for bringing this to our attention!

**Comment by benchristel** [comment]
Thanks for the reply! Unfortunately, we're stuck at v9 for now because we're using React 15. If you're willing to do the backport, that would be awesome. Otherwise, I might have time at some point during the next couple of weeks to look at it.

**Comment by jberney** [comment]
Hi @benchristel 

We tried our fix in our v9 branch, but it completely broke the iconography component. We're not going to fix this. Please update to React 16 at your earliest convenience and use PUI 12+ to get past this issue.

**Comment by benchristel** [comment]
Ok, sounds good. Thanks for the attempt :)

**Comment by jberney** [comment]
@benchristel I'm re-opening this issue. We may have a way to support React 15 in newer versions of PUI. We'll probably be exploring this next week.

**Comment by reid47** [comment]
Hi @benchristel - sorry for the delay on this, but we have just released new minor versions of PUI 10-16 that support React 16.

We determined that PUI's code does not actually require React 16 (it doesn't use any of the new features introduced in React 16+). So, we moved React/ReactDOM from our `dependencies` into `peerDependencies` and changed the versions to `^15.0.0 || ^16.0.0`. 

These are the specific PUI versions that no longer require React 16:

- `16.1.0`
- `15.1.0`
- `14.1.0`
- `13.1.0`
- `12.4.0`
- `11.2.0`
-  `10.4.0`

We hope this will make it easier for your team to upgrade PUI, since you won't need to upgrade React. I'm going to close this issue now, but feel free to reach out if you have questions!
